src/ir/ir_to_proto.h

```diff
@@ -0,0 +1,92 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#ifndef SRC_IR_IR_TO_PROTO_H_
+#define SRC_IR_IR_TO_PROTO_H_
+
+#include "src/ir/attributes/attribute.h"
+#include "src/ir/attributes/int_attribute.h"
+#include "src/ir/attributes/string_attribute.h"
+#include "src/ir/ir_context.h"
+#include "src/ir/module.h"
+#include "src/ir/proto/raksha_ir.pb.h"
+#include "src/ir/value.h"
+
+namespace raksha::ir {
+
+class IRToProto {
+ public:
+  using ID = uint64_t;
+  IRToProto(ID& id)
```

Comment:
Why passing ID by reference? Just pass them by value?

I presume we will run  multiple `IRToProto` operations in sequence? If so, we can easily initialize the second one by exposing the `last_used_id` of the first one?